catalyst_voices/apps/voices/lib/widgets/document_builder/document_property_builder_widget.dart

```diff
@@ -36,7 +37,10 @@ class DocumentPropertyBuilderWidget extends StatelessWidget {
       TokenValueCardanoADADefinition() => throw UnimplementedError(),
       DurationInMonthsDefinition() => throw UnimplementedError(),
       YesNoChoiceDefinition() => throw UnimplementedError(),
-      AgreementConfirmationDefinition() => throw UnimplementedError(),
+      AgreementConfirmationDefinition() => DocumentCheckboxBuilderWidget(
+          property: property,
+          onChanged: (val) {},
```

Comment:
Here you can pass the `onChanged` that you get via `DocumentPropertyBuilderWidget` constructor